bundles/org.openhab.binding.opengarage/src/main/java/org/openhab/binding/opengarage/internal/OpenGarageHandler.java

```diff
@@ -53,33 +53,42 @@ public class OpenGarageHandler extends BaseThingHandler {
     private final Logger logger = LoggerFactory.getLogger(OpenGarageHandler.class);
 
     private long refreshInterval;
+    private long transitionTimeSecs;
 
     private @NonNullByDefault({}) OpenGarageWebTargets webTargets;
-    private @Nullable ScheduledFuture<?> pollFuture;
+    private @Nullable VariableDelayPoller poller;
+    private Instant lastTransition;
+    private String lastTransitionText;
+
+    public @NonNullByDefault({}) String doorOpeningState;
+    public @NonNullByDefault({}) String doorOpenState;
+    public @NonNullByDefault({}) String doorClosedState;
+    public @NonNullByDefault({}) String doorClosingState;
```

Comment:
Rather replace these four lines by:
```
private OpenGarageConfiguration config = new OpenGarageConfiguration();
```
and then use config.xxx in your code.
Initialize `config` in initialize as you did.